src/bin/pgcopydb/follow.c

```diff
@@ -388,11 +388,18 @@ follow_prepare_mode_switch(StreamSpecs *streamSpecs)
 		}
 	}
 
-	/* first empty the transform queue, where a STOP message has been sent */
-	if (!stream_transform_from_queue(streamSpecs))
+	/*
+	 * If the current mode is replay, it means the previous mode was catch-up.
+	 * Therefore, we need to empty the transform queue before proceeding, where
+	 * a STOP message has been sent.
```

Comment:
That's a little confusing. I think it would be better if we would pass previousMode as well as currentMode (or maybe nextMode) to the function. At the moment we only have two modes of operations and we always go from the one to the next, but that could change and passing both args is good future-proofing. And will make this code much easier to reason about, too.

Reply:
Make sense, I'm going with `currentMode` and `previousMode`, I'm not sure how `nextMode` could fit here.

Though I have one doubt related to my change, is it guaranteed that the transform queue would be empty if say during the replay the transform process got interrupted while still have some messages in the queue?

Reply:
The transform queue is used to send LSN positions that belong to new JSON files, and triggers the transformation of an entire file at a time from JSON to SQL. In replay mode, a Unix PIPE is used in which we write the JSON messages, and the transform process reads JSON messages from the other end of the same pipe. The transform queue is never used when in replay mode.

I am not sure I got the question right, but it seems to me that the answer is “won't happen”.